libqtile/widget/yahoo_weather.py

```diff
@@ -25,53 +26,124 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
+import requests
+import json
+from requests_oauthlib import OAuth1
 from urllib.parse import urlencode
-from xml.dom import minidom
 
 from libqtile.widget import base
-from libqtile.widget.generic_poll_text import GenPollUrl
 
-QUERY_URL = 'http://query.yahooapis.com/v1/public/yql?'
-WEATHER_URL = 'http://weather.yahooapis.com/forecastrss?'
-WEATHER_NS = 'http://xml.weather.yahoo.com/ns/rss/1.0'
+# See documentation here: https://developer.yahoo.com/weather/documentation.html
+QUERY_URL = 'https://weather-ydn-yql.media.yahoo.com/forecastrss'
+APP_ID = 'xSqyTW54'
+CONSUMER_KEY = 'dj0yJmk9R0RwZ3dveWEwTHdWJmQ9WVdrOWVGTnhlVlJYTlRRbWNHbzlNQS0tJnM9Y29uc3VtZXJzZWNyZXQmc3Y9MCZ4PTVi'
+CONSUMER_SECRET = '83ea8fbd202ea06cd57fc01268139601bf966b47'
```

Comment:
It would be nice if there was a better way to distribute this, but I'm not sure there is.

Reply:
I would have no idea how. Although the app_id was specifically set up for the "qtile weatherwidget", and at the very least this way everything is transparent and editable.